### PR TITLE
ADD car test

### PR DESCRIPTION
**Pull request by tomasedl**
If you have any questions or notes, feel free to add them here or on discord

### CODE REVIEW

File: src/main/java/cz/muni/pa165/teamwhite/formula1/dao/CarDaoImpl.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/dao/CarDaoImpl.java:9]: zbytecny import?

File: src/main/java/cz/muni/pa165/teamwhite/formula1/dao/ComponentDaoImpl.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/dao/ComponentDaoImpl.java:10]: zbytecny import?

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+    @Test(expectedExceptions = ConstraintViolationException.class)
+    public void CarWithNullNameTest() {
+        Car car = new Car(null, null, null);
+
+        carDao.create(car);
+    }
+
+    @Test
+    @Rollback(value = true)
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:62]: Rollback by tu nemusel asi byt, zkus to bez nej :smile:

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
+package cz.muni.pa165.teamwhite.formula1.Dao;
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:1]: package name nemuze byt velkym pismenem, ne? :D

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        carDao.remove(carFerrari);
+
+        Assert.assertEquals(carDao.findAll().size(), 1);
+    }
+
+    public Driver createDriverLewis() {
+        Driver driver = new Driver();
+        driver.setName("Lewis");
+        driver.setSurname("Hamilton");
+        driver.setNationaliy("GB");
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:201]:
```suggestion
        driver.setNationaliry("GB");
```

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        driver.setReactions(10);
+
+        return driver;
+    }
+
+    public Driver createDriverFernando() {
+        Driver driver = new Driver();
+        driver.setName("Fernando");
+        driver.setSurname("Alonso");
+        driver.setNationaliy("ES");
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:213]:
```suggestion
        driver.setNationaliry("ES");
```

Comment (Jakub-Fajkus) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:213]: :facepalm: dyzleksye kicking in...

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 5 above)
Comment (khecova) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:201]: driver.setNationality("GB");

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 6 above)
Comment (khecova) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:213]: driver.setNationality("GB");

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+/**
+ * @author Tomas Sedlacek
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class CarTests extends AbstractTestNGSpringContextTests {
+    @PersistenceUnit
+    private EntityManagerFactory emf;
```
Comment (khecova) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:43]: nevyuzita trida

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 3 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: pushnul jsem to driv, nez jsem nasel to vlakno na stackoverflow :smiley:

File: src/main/java/cz/muni/pa165/teamwhite/formula1/dao/CarDaoImpl.java (hunk 1 above)
Comment (tomasedl) [reply src/main/java/cz/muni/pa165/teamwhite/formula1/dao/CarDaoImpl.java]: fixed

File: src/main/java/cz/muni/pa165/teamwhite/formula1/dao/ComponentDaoImpl.java (hunk 2 above)
Comment (tomasedl) [reply src/main/java/cz/muni/pa165/teamwhite/formula1/dao/ComponentDaoImpl.java]: fixed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 4 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: fixed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 5 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: fixed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 6 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: fixed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 7 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: fixed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+
+        Assert.assertEquals(carDao.findAll().size(), 1);
+    }
+
+    public Driver createDriverLewis() {
+        Driver driver = new Driver();
+        driver.setName("Lewis");
+        driver.setSurname("Hamilton");
+        driver.setNationality("GB");
+        driver.setAgresive(true);
```
Comment (khecova) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:178]: melo by byt setAggressive ... ve fci je typo 😄

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+
+        return driver;
+    }
+
+    public Driver createDriverFernando() {
+        Driver driver = new Driver();
+        driver.setName("Fernando");
+        driver.setSurname("Alonso");
+        driver.setNationality("ES");
+        driver.setAgresive(true);
```
Comment (khecova) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java:190]: setAggressive

**Review by Jakub-Fajkus — APPROVED** [review]

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 8 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: sorry, my bad. I have fixed it

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java (hunk 9 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/dao/CarTests.java]: same as above

**Review by andrlos — APPROVED** [review]
nicely done

**Review by khecova — APPROVED** [review]
nice 😄

**Pull request merged**